src/Sentry.AspNetCore/SentryHostBuilderExtensions.cs

```diff
@@ -0,0 +1,107 @@
+            this IHostBuilder builder,
+            Action<SentryAspNetCoreOptions>? configureOptions)
+            => builder.UseSentry((_, options) => configureOptions?.Invoke(options));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureOptions">The configure options.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<HostBuilderContext, SentryAspNetCoreOptions>? configureOptions)
+            => builder.UseSentry((context, sentryBuilder) =>
+                sentryBuilder.AddSentryOptions(options => configureOptions?.Invoke(context, options)));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureSentry">The Sentry builder.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<ISentryBuilder>? configureSentry) =>
+            builder.UseSentry((_, sentryBuilder) => configureSentry?.Invoke(sentryBuilder));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureSentry">The Sentry builder.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<HostBuilderContext, ISentryBuilder>? configureSentry)
+        {
+            // The earliest we can hook the SDK initialization code with the framework
+            // Initialization happens at a later time depending if the default MEL backend is enabled or not.
+            // In case the logging backend was replaced, init happens later, at the StartupFilter
+            _ = builder.ConfigureLogging((context, logging) =>
+            {
+                logging.AddConfiguration();
+
+                var section = context.Configuration.GetSection("Sentry");
+                _ = logging.Services.Configure<SentryAspNetCoreOptions>(section);
+
+                _ = logging.Services
+                    .AddSingleton<IConfigureOptions<SentryAspNetCoreOptions>, SentryAspNetCoreOptionsSetup>();
+                _ = logging.Services.AddSingleton<ILoggerProvider, SentryAspNetCoreLoggerProvider>();
+
+                _ = logging.AddFilter<SentryAspNetCoreLoggerProvider>(
+                    "Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware",
+                    LogLevel.None);
+
+                var sentryBuilder = logging.Services.AddSentry();
+                configureSentry?.Invoke(context, sentryBuilder);
+            });
+
+            _ = builder.ConfigureServices((c, s) => _ = s.AddTransient<IStartupFilter, SentryStartupFilter>());
```

Comment:
TryAddTransient is also an issue here, if any IStartupFilter was already added (which most likely it will) this will not be registered

Reply:
Something lke:

```
            if (!s.Any(d =>  d.ServiceType == descriptor.ServiceType &&    d.GetImplementationType() == implementationType))

```

Reply:
Oh nice, I didn't think of that.
Just ran a quick test and yes that would work.